File: clients/console-client/src/main/java/ec/edu/grupo3/client/controllers/AuthController.java
package ec.edu.grupo3.client.controllers;

import ec.edu.grupo3.client.models.UserSession;
import ec.edu.grupo3.client.views.AuthView;

public class AuthController {
    
    private static final String VALID_USER = "MONSTER";
    private static final String VALID_PASS = "MONSTER9";

    private final AuthView authView = new AuthView();

    public UserSession authenticate() {
        while (true) {
            authView.showBanner();
            String user = authView.getUsername();
            String pass = authView.getPassword();

            if (VALID_USER.equals(user.trim()) && VALID_PASS.equals(pass)) {
                authView.showSuccess(user.trim());
                return new UserSession(user.trim(), true);
            } else {
                authView.showError("Credenciales incorrectas. Verifique su usuario y contraseña.");
            }
        }
    }
}

File: dotnet-clients/web-client/Controllers/AuthController.cs
using System.Security.Claims;
using Microsoft.AspNetCore.Authentication;
using Microsoft.AspNetCore.Authentication.Cookies;
using Microsoft.AspNetCore.Mvc;
using web_client.Models;

namespace web_client.Controllers
{
    public class AuthController : Controller
    {
        private const string VALID_USER = "MONSTER";
        private const string VALID_PASS = "MONSTER9";

        [HttpGet]
        public IActionResult Login()
        {
            if (User.Identity!.IsAuthenticated)
            {
                return RedirectToAction("Index", "Conversion");
            }
            return View(new LoginViewModel());
        }

        [HttpPost]
        [ValidateAntiForgeryToken]
        public async Task<IActionResult> Login(LoginViewModel model)
        {
            if (!ModelState.IsValid)
            {
                return View(model);
            }

            if (model.Username.Trim() == VALID_USER && model.Password == VALID_PASS)
            {
                var claims = new List<Claim>
                {
                    new Claim(ClaimTypes.Name, model.Username.Trim()),
                    new Claim(ClaimTypes.Role, "Administrator")
                };

                var claimsIdentity = new ClaimsIdentity(claims, CookieAuthenticationDefaults.AuthenticationScheme);

                var authProperties = new AuthenticationProperties
                {
                    AllowRefresh = true,
                    IsPersistent = true
                };

                await HttpContext.SignInAsync(
                    CookieAuthenticationDefaults.AuthenticationScheme,
                    new ClaimsPrincipal(claimsIdentity),
                    authProperties
                );

                return RedirectToAction("Index", "Conversion");
            }

            model.ErrorMessage = "Usuario o contraseña incorrectos. Por favor, verifique sus credenciales.";
            return View(model);
        }

        [HttpGet]
        public async Task<IActionResult> Logout()
        {
            await HttpContext.SignOutAsync(CookieAuthenticationDefaults.AuthenticationScheme);
            return RedirectToAction("Login", "Auth");
        }
    }
}